at_lookup/test/outbound_message_listener_test.dart

```diff
@@ -177,4 +177,65 @@ void main() {
       expect(response, 'stream:@bob:phone@alice');
     });
   });
+
+  group('A group of tests to verify AtTimeOutException', () {
```

Comment:
Good work! Main thing missing now is tests where we start a read, then we send data in chunks to the messageHandler but with delays ('await Future.delayed(Duration(xxx))`) of various lengths in between, and check both positive and negative scenarios over longer durations

Reply:
added two tests
1. A test to verify full response received - delay between messages from server
2. A test to verify timeout - delay between messages from server
Please check whether this is inline with what you expected.
Will try to add more tests.

Reply:
Looks good!